spec/models/user_spec.rb

```diff
@@ -221,6 +223,18 @@
       end
     end
 
+    context 'public_keys' do
+      it 'returns correct public keys' do
+        expect(subject.public_keys).to match_array([public_key1, public_key2])
+      end
+
+      it 'destroys the public keys on destroy' do
```

Comment:
`deletes the public keys on destroy`
You don't destroy them (and don't check it here) because you only delete the rows and do not load the objects from the database and call their destroy method.